### PR TITLE
Save attribute checkbox state

### PR DESCRIPTION
**Pull request by KTibow**
## Proposed change

Save whether or not the attribute checkbox is checked in localStorage. This can help if you have a lot of entities, so you don't have to wait forever to load the attributes.
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
# Go into the states tab of devtools
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: #7537
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by spacegaier — CHANGES_REQUESTED** [review]
Two points:
1. I propose to rename the storage parameter, so that the area/topic comes first: "devToolsShowAttributes". That would more closely match existing stored parameters such as "sidebarPanelOrder".
2. There is already coding in HA how to handle local storage parameters. That should perhaps be reused. See https://github.com/home-assistant/frontend/blob/c1965492d99654ef5c43267cf2917c4ded885134/src/components/ha-sidebar.ts#L191-L194

File: src/panels/developer-tools/state/developer-tools-state.js [hunk 1]
```diff
@@ -459,6 +459,10 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
     return Array.isArray(value) ? value.join(", ") : value;
   }
 
+  saveAttributeCheckboxState(ev) {
+    localStorage.setItem("showAttributesDevTools", ev.target.checked);
```
Comment (KTibow) [inline src/panels/developer-tools/state/developer-tools-state.js:463]:
```suggestion
    localStorage.setItem("devToolsShowAttributes", ev.target.checked);
```

File: src/panels/developer-tools/state/developer-tools-state.js [hunk 2]
```diff
@@ -275,7 +275,7 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
 
       _showAttributes: {
         type: Boolean,
-        value: true,
+        value: JSON.parse(localStorage.getItem("showAttributesDevTools") || true),
```
Comment (KTibow) [inline src/panels/developer-tools/state/developer-tools-state.js:278]:
```suggestion
        value: JSON.parse(localStorage.getItem("devToolsShowAttributes") || true),
```

**Comment by KTibow** [comment]
@spacegaier Could you point me how to use the localStorage decorator? I skimmed through the code, but I couldn't figure out the parameter order.

**Comment by spacegaier** [comment]
> 
> 
> @spacegaier Could you point me how to use the localStorage decorator? I skimmed through the code, but I couldn't figure out the parameter order.

What do you mean by "property order"? Never used local storage in HA myself.

**Comment by KTibow** [comment]
I mean how to use it. I can see that the first thing you pass to `@LocalStorage` is the name of the property, but I don't know what everything else is.

**Comment by spacegaier** [comment]
It's all defined in the HA coding: https://github.com/home-assistant/frontend/blob/c1965492d99654ef5c43267cf2917c4ded885134/src/common/decorators/local-storage.ts#L85-L89

Second parameter seems to indicate whether it's a property and if it is, then the value is automatically read from it by the looks of it. So it is basically subscribing to value changes (changes introduced here https://github.com/home-assistant/frontend/pull/6776).

**Comment by bramkragten** [comment]
This is not a typescript file and not a Lit element, so you cant use that decorator without converting it first.

File: src/panels/developer-tools/state/developer-tools-state.js [hunk 3]
```diff
@@ -459,6 +459,10 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
     return Array.isArray(value) ? value.join(", ") : value;
   }
 
+  saveAttributeCheckboxState(ev) {
+    localStorage.setItem("devToolsShowAttributes", ev.target.checked);
```
Comment (bramkragten) [inline src/panels/developer-tools/state/developer-tools-state.js:463]: This could throw (in safari private mode I think)

Comment (KTibow) [reply src/panels/developer-tools/state/developer-tools-state.js]: I think I fixed it.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**